exe/CMakeLists.txt

```diff
@@ -12,13 +12,29 @@ include_directories(
     ../lib/inc # the libfacter headers
     ${Boost_INCLUDE_DIRS}
     ${YAMLCPP_INCLUDE_DIRS}
+    ${LEATHERMAN_RUBY_INCLUDE}
+    ${LEATHERMAN_FILE_UTIL_INCLUDE}
     ${LEATHERMAN_UTIL_INCLUDE}
     ${LEATHERMAN_NOWIDE_INCLUDE}
+    ${CPPHOCON_INCLUDE_DIRS}
+)
+
+link_directories(
+    ${CPPHOCON_LIBRARY_DIRS}
 )
 
 add_executable(facter ${FACTER_SOURCES})
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+    target_link_libraries(facter rt)
+endif()
+
 target_link_libraries(facter libfacter
+    ${CPPHOCON_LIBRARIES}
     ${Boost_PROGRAM_OPTIONS_LIBRARY}
+    ${Boost_THREAD_LIBRARY}
+    ${LEATHERMAN_RUBY_LIB}
+    ${LEATHERMAN_FILE_UTIL_LIB}
```

Comment:
Windows build is failing because leatherman_file_util has a dependency on boost_filesystem. You'll want to put the boost_filesystem dependency after leatherman_file_util. I'd also suggest moving the other boost libraries after leatherman libraries, as boost won't depend on leatherman.

There are certain scenarios where that suggestion can cause trouble - dynamic leatherman but static boost, and leatherman only importing some of the functions from boost, can result in some symbols binding to leatherman dlls and others pulling them in from the static boost libraries; in that case you may have problems with global static members, where there are two copies of the same global and different functions refer to different versions. Hence don't use dynamic leatherman and static boost.